doc/migrations/v0.43-v0.44.md

```diff
@@ -0,0 +1,117 @@
+# Migrating to libp2p@44 <!-- omit in toc -->
+
+A migration guide for refactoring your application code from libp2p v0.43.x to v0.44.0.
+
+## Table of Contents <!-- omit in toc -->
+
+- [Connection Manager](#connection-manager)
+- [Connection Gater](#connection-gater)
+
+## Connection Manager
+
+The connection manager has been refactored to be simpler and faster.  The changes
+are mostly internal but some configuration options have changed.
+
+**Before**
+
+```js
+import { createLibp2p } from 'libp2p'
+
+const node = await createLibp2p({
+  connectionManager: {
+    // this option has been removed as auto dial now runs in response to peers
+    // disconnecting and is no longer on a timer
+    autoDialInterval: 0,
+
+    // this has been renamed to `maxPeerAddrsToDial`
+    maxAddrsToDial: 20,
+
+    // this has been renamed to `maxParallelDialsPerPeer`
+    maxDialsPerPeer: 20,
+
+    // these should now be ranges expressed as MultiaddrFilters
+    allow: [
+      '/ip4/0.0.0.0/tcp/123'
+    ],
+
+    // these should now be ranges expressed as MultiaddrFilters
+    deny: [
+      '/ip4/0.0.0.0/tcp/123'
+    ]
+  }
+})
+```
+
+**Before**
+
+```js
+import { createLibp2p } from 'libp2p'
+import { MultiaddrFilter } from from '@multiformats/multiaddr'
+
+const node = await createLibp2p({
+  connectionManager: {
+    // how many peers to dial at once while trying to ensure the node
+    // is above minConnections
+    autoDialConcurrency: 25,
+
+    // a low value allows user-initiated dials to take priority over
+    // auto dials
+    autoDialPriority: 0,
+
+    // this was previously named `maxAddrsToDial`
+    maxPeerAddrsToDial: 20,
+
+    // this was previously named `maxDialsPerPeer`
+    maxParallelDialsPerPeer: 20,
```

Comment:
Just so i understand fully, this setting changes whether we can attempt to dial 2 or 20 multiaddrs at once when attempting to connect to a specific peer, correct?

Would 'maxParallelMultiAddrDialsPerPeer' be a more explicit name? Is there a way we can get that explicitness without the verbosity?

Reply:
We only dial multiaddrs so maybe having `MultiAddr` in the name is redundant?

Reply:
But yes, a setting of `2` with a peer that has `10` addresses means we'll dial all 10 addresses but only ever 2 at once.  This stops a peer with a lot of spam addresses blocking the dial queue.